packages/flutter/lib/src/widgets/spell_check.dart

```diff
@@ -0,0 +1,322 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/services.dart'
+    show SpellCheckResults, SpellCheckService, SuggestionSpan, TextEditingValue;
+
+/// Controls how spell check is performed for text input.
+///
+/// This configuration determines the [SpellCheckService] used to fetch the
+/// [List<SuggestionSpan>] spell check results and the
+/// [SpellCheckSuggestionsHandler] used to mark and display replacement
+/// suggestions for misspelled words within text input.
+class SpellCheckConfiguration {
+  /// Creates a configuration that specifies the service and suggestions handler
+  /// for spell check.
+  const SpellCheckConfiguration({
+    this.spellCheckService,
+    this.spellCheckSuggestionsHandler,
+    this.misspelledTextStyle,
+  });
+
+  /// The service used to fetch spell check results for text input.
+  final SpellCheckService? spellCheckService;
+
+  /// The handler used to mark misspelled words in text input and display
+  /// a menu of the replacement suggestions for these misspelled words.
+  final SpellCheckSuggestionsHandler? spellCheckSuggestionsHandler;
+
+  /// Style used to indicate misspelled words.
+  ///
+  /// This is nullable to allow style-specific wrappers of [EditableText]
+  /// to infer this, but this must be specified if this configuration is
+  /// provided directly to [EditableText] or its construction will fail with an
+  /// assertion error.
+  final TextStyle? misspelledTextStyle;
+
+  /// A [SpellCheckConfiguration] that indicates a text input widget does not
+  /// wish to enable spell check.
+  static const SpellCheckConfiguration disabled = SpellCheckConfiguration(
+    spellCheckService: null,
+    spellCheckSuggestionsHandler: null,
+    misspelledTextStyle: null,
+  );
+
+  /// Returns a copy of the current [SpellCheckConfiguration] instance with
+  /// specified overrides.
+  SpellCheckConfiguration copyWith({
+    SpellCheckService? spellCheckService,
+    SpellCheckSuggestionsHandler? spellCheckSuggestionsHandler,
+    TextStyle? misspelledTextStyle}) {
+    return SpellCheckConfiguration(
+      spellCheckService: spellCheckService ?? this.spellCheckService,
+      spellCheckSuggestionsHandler:
+        spellCheckSuggestionsHandler ?? this.spellCheckSuggestionsHandler,
+      misspelledTextStyle: misspelledTextStyle ?? this.misspelledTextStyle,
+    );
+  }
+}
+
+/// Determines how misspelled words are indicated in text input and how
+/// replacement suggestions for misspelled words are displayed via menu.
+mixin SpellCheckSuggestionsHandler {
```

Comment:
Should this be an `abstract class` instead?

Reply:
@justinmc and I discussed this here https://github.com/flutter/flutter/pull/96629#discussion_r855648512. We talked more offline and since we had no strong reasons to make it an `abstract class` over a `mixin`, I left it.